src/core/include/openvino/core/any.hpp

```diff
@@ -423,7 +423,7 @@ class OPENVINO_API Any {
     template <class U>
     [[noreturn]] static typename std::enable_if<!EqualityComparable<U>::value, bool>::type equal_impl(const U&,
                                                                                                       const U&) {
-        OPENVINO_THROW("Could not compare types without equality operator");
+        OPENVINO_THROW_NORETURN("Could not compare types without equality operator");
```

Comment:
maybe of ov::Any is so huge because of noreturn exceptions?
It's possible to provide this functionality without exceptions?

Reply:
at least  we can try to remove this exceptions and see whether it affects binary size